OneThreadSolution.java

```diff
@@ -0,0 +1,61 @@
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Created by user on 02.04.2016.
+ */
+public class OneThreadSolution {
+
+    private boolean hasDifferentDigits(long l) {
+        int digit;
+        Set<Integer> digits = new HashSet<>();
+        digits.add(0);
+        while (l > 0) {
```

Comment:
If I make a method here, then after reuse I'll need additional loops. Is it really better than this variant?